### ISSUE TITLE
EMDPFit not fitting, the .nml file may need to be fixed

### ISSUE THREAD

**Issue by fulinwang** [marcdegraef/EMsoft#10]
Firstly, the start fit function in Workbench by BlueQuartz is not functioning. So I have to use EMDPFit.

In the `EMSPFit.nml` file, some parameters are undefined. This is found when running `EMDPFit`, an error is displayed, for example, `Cannot match nameless object name energyfile`. When I looked into the `NameListHandlers.f90`, there are subroutines `GetEMDPFit4NameList` and `GetEMDPFitNameList`. And in the latter (`GetEMDPFitNameList`), energyfile is indeed not defined as a parameter. After deleting this option in the .nml file and some other ambiguous parameters. `EMDPFit` can run. The parameters that need to be changed are:
- change `exptfile_pat1` to `exptfile`, and delete exptfile_pat2 (and 3, 4).
- change `phi1_pat1` to `phi1`, the same goes for phi and phi2, and delete the rest with _pat2 ( and 3, 4).
- change `stepx` and `stepy` to `step_xpc` and `step_ypc`.

Although EMDPFit runs after changing the .nml file, it seems not fitting. The display of "`Least value of F`" is always NaN. I think probably `bobyqa.f90` needs to be modified, since F is computed there.

Either Workbench or EMDPFit needs to be fixed so we can fit the pattern center.

Thanks.

**Comment by CiosG** [comment]
Dear Saransh,
I have tried the EMDPFit by generating 640x480 pattern with euler angles of 0,0,0 for FCC, L - 17408, xpc  = 3.2,  ypc = 32.64 in EMEBSD and putting it into EMDPFit, 
starting values for: 
euler angles were 0,0,0 
L - 17400, 
xpc  = 3.2,  ypc = 32.64 

and steps were:
 step_xpc 	= 1, 
 step_ypc 	= 1,
 step_L 	= 10.0,
 step_phi1 = 1,
 step_phi 	= 1,
 step_phi2 = 1,

and I have obtained after 5 runs:
X-PC: 3.2385....
Y-PC: 32.561...
L = 17049.94
phi1,PHI, phi2 0.003394, -0.142767, -0.042158

I have tried few combinations and always largest error is in L and more runs.
I have also tried case when after first step the theoretical value should be obtained but never was.
Omega angle option in EMDPFit is missing (for my setup should be 180).
Do you have IDL GUI version for windows?

**Comment by saransh13** [comment]
Apologies for the delay in replying. I had been traveling for a conference
the last week.
I've tried the same test with a different set of parameters and the results
seem to converge very close to the correct values.
How was your experimental data file formatted? I have a feeling that the
experimental and simulated patterns have a different origin?
Can you post your data so I can try and reproduce these numbers?

Thanks,
Saransh

**Comment by CiosG** [comment]
Dear Saransh,
it was my mistake. I am sorry. I had badly formatted data but now I have matlab program from prof. M. DeGraef that converts images to binary.  Now the program was fitting parameters however I used binned  raw (with background) pattern 120x120 and the program worked but the numbers were far from good. I am sure it is because of 4x4 binning and uneven background due to TKD geometry. I just used my old data to play with your software and until now I really liked it. I hope I will succeed with EMDPFit and next step will be DI for EBSD and TKD.

Best regards
Grzegorz
